src/Features/Core/Portable/LanguageServices/SymbolDisplayService/AbstractSymbolDisplayService.AbstractSymbolDescriptionBuilder.cs

```diff
@@ -151,48 +144,31 @@ protected SemanticModel GetSemanticModel(SyntaxTree tree)
                 return null;
             }
 
-            protected Compilation GetCompilation()
+            protected Compilation Compilation
                 => _semanticModel.Compilation;
 
             private async Task AddPartsAsync(ImmutableArray<ISymbol> symbols)
             {
                 var firstSymbol = symbols[0];
+
+                // Grab the doc comment once as computing it for each portion we're concatenating can be expensive for
+                // LSIF (which does this for every symbol in an entire solution).
+                var firstSymbolDocumentationComment = firstSymbol.GetAppropriateDocumentationComment(Compilation, CancellationToken);
+
                 await AddDescriptionPartAsync(firstSymbol).ConfigureAwait(false);
 
                 AddOverloadCountPart(symbols);
                 FixAllStructuralTypes(firstSymbol);
-                AddExceptions(firstSymbol);
+                AddExceptions(firstSymbolDocumentationComment);
                 AddCaptures(firstSymbol);
 
-                AddDocumentationContent(firstSymbol);
+                AddDocumentationContent(firstSymbol, firstSymbolDocumentationComment);
             }
 
-            private void AddDocumentationContent(ISymbol symbol)
+            private void AddDocumentationContent(ISymbol symbol, DocumentationComment documentationComment)
             {
-                var formatter = Services.GetRequiredLanguageService<IDocumentationCommentFormattingService>(_semanticModel.Language);
```

Comment:
This logic was alreayd duplicated in ISymbolExtensions_2.cs, so it's unified now.